Src/ILGPU.Algorithms.Tests/XMathTests.cs

```diff
@@ -35,8 +35,17 @@ private class FloatPrecisionComparer
             public FloatPrecisionComparer(uint decimalPlaces) =>
                 Margin = MathF.Pow(10, -decimalPlaces);
 
-            public override bool Equals(float x, float y) =>
-                Math.Abs(x - y) < Margin;
+            public override bool Equals(float x, float y)
+            {
+                if ((float.IsNaN(x) && float.IsNaN(y)) ||
+                    (float.IsPositiveInfinity(x) && float.IsPositiveInfinity(y)) ||
+                    (float.IsNegativeInfinity(x) && float.IsNegativeInfinity(y)))
+                {
+                    return true;
```

Comment:
What happen in cases in which `x = +Infinity` and `y = -Infinity`? Does this ever happen? If yes, is this case important to capture?

Reply:
It's not immediately obvious, even to me, so I'll fix it up to be explicit.

When you subtract from opposite infinities, it does not change:
`+Infinity` - `-Infinity` = `+Infinity`
`-Infinity` - `+Infinity` = `-Infinity`

When you add opposite infinities, it changes:
`+Infinity` + `-Infinity` = `NaN`
`-Infinity` + `+Infinity` = `NaN`

We only use subtraction in the comparer, so the current code just happens to work...

Reply:
Yeah, you're right. It is always confusing when arguing about 'NaN's' and 'Infinity' float values 🔢